Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+    private static CloudPool CreatePool(BatchClient client, String poolId) throws BatchErrorException, IllegalArgumentException, IOException, InterruptedException, TimeoutException {
+        // Pool only have 1 A1 VM
+        String osPublisher = "OpenLogic";
+        String osOffer = "CentOS";
+        String poolVMSize = "STANDARD_A1";
+        int poolVMCount = 1;
+        long POOL_STEADY_TIMEOUT = 5 * 60 * 1000;
+        long VM_READY_TIMEOUT = 20 * 60 * 1000;
+
+        // Check if pool exists
+        if (!client.getPoolOperations().existsPool(poolId)) {
+
+            // See detail of creating IaaS pool at https://blogs.technet.microsoft.com/windowshpc/2016/03/29/introducing-linux-support-on-azure-batch/
+            // Get the sku image reference
+            List<NodeAgentSku> skus = client.getAccountOperations().listNodeAgentSkus();
+            String skuId = null;
+            ImageReference imageRef = null;
+
+            for (NodeAgentSku sku : skus) {
+                if (sku.getOsType() == OSType.LINUX) {
+                    for (ImageReference imgRef : sku.getVerifiedImageReferences()) {
+                        if (imgRef.getPublisher().equalsIgnoreCase(osPublisher) && imgRef.getOffer().equalsIgnoreCase(osOffer)) {
+                            imageRef = imgRef;
+                            skuId = sku.getId();
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // Use IaaS VM with Linux
+            VirtualMachineConfiguration configuration = new VirtualMachineConfiguration();
+            configuration.setNodeAgentSKUId(skuId);
+            configuration.setImageReference(imageRef);
+
+            client.getPoolOperations().createPool(poolId, poolVMSize, configuration, poolVMCount);
+        }
+
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+        boolean steady = false;
+
+        // Wait for a maximum of 5 minutes for the VM to be allocated
+        while (elapsedTime < POOL_STEADY_TIMEOUT) {
+            CloudPool pool = client.getPoolOperations().getPool(poolId);
+            if (pool.getAllocationState() == AllocationState.STEADY) {
+                steady = true;
+                break;
+            }
+            System.out.println("wait 30 seconds for pool steady...");
+            Thread.sleep(30 * 1000);
+            elapsedTime = (new Date()).getTime() - startTime;
+        }
+
+        if (!steady) {
+            throw new TimeoutException("The pool did not reach a stead state in the allotted time");
+        }
+
+        // The VMs in the pool don't need to be in and IDLE state in order to submit a job.
+        // The following code is just an example of how to poll for the VM state
+        startTime = System.currentTimeMillis();
+        elapsedTime = 0L;
+        boolean hasIdleVM = false;
+
+        // Wait for a maximum of 20 minutes for at least 1 VM to reach the IDLE state
```

Comment:
again, don't mention the 20m, instead just said "wait for VM_READY_TIMEOUT for at least 1 VM to reach the idle state" or "wait for at least 1 VM to reach the idle state" -- they can read the code to see the amount of time.